### PR TITLE
Add: audit event to GET /resources/:account/:kind/*identifier

### PR DESCRIPTION
**Pull request by yoavgeva**
### Desired Outcome
This PR adds an audit log message for get resource using the API endpoint `GET /resources/:account/:kind/*identifier`

### Implemented Changes

*Describe how the desired outcome above has been achieved with this PR. In
particular, consider:*

- Example audit message

**Successful audit:**
```
<86>1 2023-01-01T10:19:59.551Z - conjur 5979 resource [auth@43868 user="cucumber:user:alice"][subject@43868 resource="cucumber:variable:@namespace@/app-01.mycorp.com"][client@43868 ip="127.0.0.1"][action@43868 result="success" operation="get"] cucumber:user:alice successfully fetched resource details.
```

**Failed audit:**
```
<84>1 2023-01-01T09:45:03.798Z - conjur 92353 resource [auth@43868 user="cucumber:user:alice"][subject@43868 resource="cucumber:santa:claus"][client@43868 ip="127.0.0.1"][action@43868 result="failure" operation="get"] cucumber:user:alice failed to fetch resource details.: Santa 'claus' not found in account 'cucumber'
```

### Connected Issue/Story

Resolves [ONYX-27697](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-27697)

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [x] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

File: app/controllers/resources_controller.rb [hunk 1]
```diff
@@ -60,7 +60,15 @@ def index
...
-    render(json: resource)
+    begin
+      result = resource
+      audit_show_success
+    rescue => e
+      audit_show_failure(e.message)
+      raise e
+    end
+
+    render(json: result)
```
Comment (aloncarmel111) [inline app/controllers/resources_controller.rb:71]: return should be only for success response

File: app/models/audit/event/show.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+      def message
+        attempted_action.message(
+          success_msg: "#{@user_id} successfully fetched #{message_id} details.",
+          failure_msg: "#{@user_id} failed to fetch #{message_id} details",
+          error_msg: @error_message
+        )
+      end
+
+      def message_id
```
[the remark this answers is not in the document]
Comment (jvanderhoof) [reply app/models/audit/event/show.rb:55]: (Nit) I agree with the above recommendation.  This method can be removed by adding:

```ruby
attr_reader :message_id
```
to the top of this class.  The above will create the `@message_id` that's set in the initializer.

**Review by jvanderhoof — APPROVED** [review]
Overall this looks solid.  Thanks for taking care of this gap @yoavgeva!

**Review by jvanderhoof — APPROVED** [review]
LGTMed!

**Review by jvanderhoof — APPROVED** [review]

**Pull request merged**